anchore-ci/common_tasks

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env bash
+
+############################################################
+# Common CI tasks. These are intended to be reusable across
+# projects, and easily overriden.
+############################################################
+
+clean() {
+  TEST_IMAGE_NAME="${1:?'Missing required parameter: TEST_IMAGE_NAME.'}"
+
+  print_colorized WARN "Deleting virtualenv, cache, build and testing temp data."; echo
+
+  continue_prompt
+  rm -rf venv ./*.egg-info dist build .tox
+  find . \( -type f -name "*.py[co]" -o -type d -name "__pycache__" \) -delete
+
+  print_colorized WARN "Deleting Docker image '${TEST_IMAGE_NAME}.'"; echo
+
+  continue_prompt
+  docker rmi "${TEST_IMAGE_NAME}"
+
+  print_colorized INFO "Finished clean."; echo
+}
+
+lint() {
+    hash tox || python -m pip install -q tox
+
+    print_colorized INFO "Linting code."; echo
+
+    tox -e flake8
+
+    print_colorized INFO "Finished linting code."; echo
+}
+
+test-unit() {
+    # Declare default tox envs, but allow TOX_ENV to override
+    default_tox_envs="py27 py36"
+    tox_env="${TOX_ENV:-$default_tox_envs}"
+
+    hash tox || python -m pip install -q tox
+
+    print_colorized INFO "Running unit tests."; echo
+
+    for e in $(echo "${tox_env}"); do
+        print_colorized INFO "Running unit tests using tox env $e."
+        tox tests/unit -e "$e" --result-json .tox/unit-test-report.log
+
+        print_colorized INFO "Finished running unit tests using tox env $e."
+    done
+
+    print_colorized INFO "Finished running unit tests."; echo
+}
+
+test-functional() {
+    # Declare default tox envs, but allow TOX_ENV to override
+    default_tox_envs="py36-anchore_latest py36-anchore_0.6.0 py36-anchore_0.5.1"
+    tox_env="${TOX_ENV:-$default_tox_envs}"
+
+    hash tox || python -m pip install -q tox
+
+    print_colorized INFO "Running functional tests."; echo
+
+    for e in $(echo "${tox_env}"); do
+        print_colorized INFO "Running functional tests using tox env $e."
+        tox -c tests/functional/tox.ini -e "$e" --result-json .tox/functional-test-report.log
+        print_colorized INFO "Finished running functional tests using tox env $e."
+    done
+
+    print_colorized INFO "Finished running functional tests."; echo
+}
+
+push-dev-image() {
```

Comment:
I'd prefer release tasks in their own script/library. I think it's a good organizational distinction between the tasks and this library will become quite large as we add more CI functionality to it.

Reply:
I dig it. I'll split it up.